order/interface.go

```diff
@@ -43,6 +44,16 @@ const (
 	// to use lease durations outside of the default/legacy 2016 block
 	// duration.
 	VersionLeaseDurationBuckets Version = 2
+
+	// VersionSidecarChannel is the order version that added sidecar
```

Comment:
Hmm, so is ths idea that the bid order itself will contain a sidecar ticket rather than open encoding some of the fields? As an example, a `push_amt` field could've been added to the bid. However, I think this is fine as if someone wants to create a balanced channel (so they're the receiver and offerer of the sidecar channel), we can have most of the interaction be hidden behind the RPC API. 

This brings to light that we should also allow that self balanced channel use case to be accessible over the RPC API via just the regular `SubmitOrder(bid)` call. The only changes we'd need here would be do specify the push amt manually (we should call it something else, like `self_chan_balance`) or something, and then update the acceptance logic to ensure that the push amt is set within the channel itself. This might either require a channel acceptor hook, or this information might already be available (or can be added) within the pending channels call itself.

Reply:
I ended up creating an independent PR for the `self_chan_balance`. At first I wasn't sure whether the feature was interesting outside of sidecar orders. Since it apparently clearly is, I added that PR first and created a 3 part split of the PRs.
The new user flow allows `self_chan_balance` to be used outside of sidecar channels (by just specifying the `--self_chan_balance` when submitting a bid order). When using a sidecar ticket the ticket itself will just dictate (=overwrite) that self channel balance when submitting a bid order.